standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/DatabaseProduct.java

```diff
@@ -749,6 +753,22 @@ public Object getBoolean(boolean val) {
     return val;
   }
 
+  public Object convertDateValue(Object dateValue) {
+    assert dateValue instanceof String;
```

Comment:
assert are ignored in production code

Reply:
exactly, this is for verifying the input in the unit tests